File: check-all-tables.py
#!/usr/bin/env python3
"""
Script to check all required tables and columns for the data replacement
"""

import os
from supabase import create_client, Client
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Configuration
SUPABASE_URL = os.getenv('PROJECT_URL', 'https://zdnyhzasvifrskbostgn.supabase.co')
SUPABASE_SERVICE_ROLE_KEY = os.getenv('SERVICE_ROLE_KEY', '')

def check_all_tables():
    """Check all required tables and columns"""
    print("Checking all required tables and columns...")
    
    try:
        # Initialize Supabase client
        supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY)
        
        # Tables that should exist
        required_tables = [
            'members', 'users', 'financial_years', 'system_settings',
            'member_balances', 'contributions', 'transactions', 'loans',
            'interest_accruals', 'audit_logs'
        ]
        
        print("\n=== CHECKING REQUIRED TABLES ===")
        for table in required_tables:
            try:
                response = supabase.table(table).select('*').limit(1).execute()
                print(f"✓ {table}: EXISTS")
            except Exception as e:
                if "Could not find" in str(e):
                    print(f"✗ {table}: MISSING")
                else:
                    print(f"✗ {table}: ERROR - {e}")
        
        # Check members table columns specifically
        print("\n=== CHECKING MEMBERS TABLE COLUMNS ===")
        try:
            response = supabase.table('members').select('*').limit(1).execute()
            if response.data:
                print("Current members table columns:")
                for key in response.data[0].keys():
                    print(f"  - {key}")
                
                # Check for specific required columns
                required_columns = ['name', 'status', 'membership_fee', 'closing_balance', 'share_value', 'date_joined']
                print(f"\nChecking required columns:")
                for col in required_columns:
                    if col in response.data[0]:
                        print(f"✓ {col}: EXISTS")
                    else:
                        print(f"✗ {col}: MISSING")
            else:
                print("No members found in table")
        except Exception as e:
            print(f"Error reading members table: {e}")
        
        # Check financial_years table structure
        print("\n=== CHECKING FINANCIAL_YEARS TABLE ===")
        try:
            response = supabase.table('financial_years').select('*').limit(1).execute()
            if response.data:
                print("Financial years table columns:")
                for key in response.data[0].keys():
                    print(f"  - {key}")
            else:
                print("No financial years found (table may be empty)")
        except Exception as e:
            print(f"Error reading financial_years table: {e}")
            
        return True
        
    except Exception as e:
        print(f"Error checking tables: {e}")
        return False

if __name__ == "__main__":
    print("Comprehensive Database Schema Check")
    print("=" * 50)
    check_all_tables()

File: check_all_tables.js
const { createClient } = require('@supabase/supabase-js');

// Use the same values from src/config/supabase.ts
const supabaseUrl = 'https://zdnyhzasvifrskbostgn.supabase.co';
const supabaseAnonKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InpkbnloemFzdmlmcnNrYm9zdGduIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTgwMjQ0ODQsImV4cCI6MjA3MzYwMDQ4NH0.s_AXhoRM9tV4F166Bhd5fG7Z14kLA0iz0l08dlzZvnM';

const supabase = createClient(supabaseUrl, supabaseAnonKey);

async function checkAllTables() {
  console.log('Checking all tables in database...\n');
  
  try {
    // Try to get information about all tables by querying information_schema
    // Note: This might not work with Supabase's RLS, so we'll try a different approach
    
    console.log('1. Checking for member_balances table:');
    const { data: balancesData, error: balancesError } = await supabase
      .from('member_balances')
      .select('*')
      .limit(1);
    
    if (balancesError) {
      console.log('  member_balances table error:', balancesError.message);
    } else if (balancesData && balancesData.length > 0) {
      console.log('  member_balances table exists with', balancesData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(balancesData[0]));
    } else {
      console.log('  member_balances table exists but is empty or no rows returned');
    }
    
    console.log('\n2. Checking for contributions table:');
    const { data: contributionsData, error: contributionsError } = await supabase
      .from('contributions')
      .select('*')
      .limit(1);
    
    if (contributionsError) {
      console.log('  contributions table error:', contributionsError.message);
    } else if (contributionsData && contributionsData.length > 0) {
      console.log('  contributions table exists with', contributionsData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(contributionsData[0]));
    } else {
      console.log('  contributions table exists but is empty or no rows returned');
    }
    
    console.log('\n3. Checking for transactions table:');
    const { data: transactionsData, error: transactionsError } = await supabase
      .from('transactions')
      .select('*')
      .limit(1);
    
    if (transactionsError) {
      console.log('  transactions table error:', transactionsError.message);
    } else if (transactionsData && transactionsData.length > 0) {
      console.log('  transactions table exists with', transactionsData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(transactionsData[0]));
    } else {
      console.log('  transactions table exists but is empty or no rows returned');
    }
    
    console.log('\n4. Checking for financial_years table:');
    const { data: financialYearsData, error: financialYearsError } = await supabase
      .from('financial_years')
      .select('*')
      .limit(1);
    
    if (financialYearsError) {
      console.log('  financial_years table error:', financialYearsError.message);
    } else if (financialYearsData && financialYearsData.length > 0) {
      console.log('  financial_years table exists with', financialYearsData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(financialYearsData[0]));
    } else {
      console.log('  financial_years table exists but is empty or no rows returned');
    }
    
    console.log('\n5. Checking for settings table:');
    const { data: settingsData, error: settingsError } = await supabase
      .from('settings')
      .select('*')
      .limit(1);
    
    if (settingsError) {
      console.log('  settings table error:', settingsError.message);
    } else if (settingsData && settingsData.length > 0) {
      console.log('  settings table exists with', settingsData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(settingsData[0]));
    } else {
      console.log('  settings table exists but is empty or no rows returned');
    }
    
    console.log('\n6. Checking for users table:');
    const { data: usersData, error: usersError } = await supabase
      .from('users')
      .select('*')
      .limit(1);
    
    if (usersError) {
      console.log('  users table error:', usersError.message);
    } else if (usersData && usersData.length > 0) {
      console.log('  users table exists with', usersData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(usersData[0]));
    } else {
      console.log('  users table exists but is empty or no rows returned');
    }
    
    console.log('\n7. Checking for profiles table:');
    const { data: profilesData, error: profilesError } = await supabase
      .from('profiles')
      .select('*')
      .limit(1);
    
    if (profilesError) {
      console.log('  profiles table error:', profilesError.message);
    } else if (profilesData && profilesData.length > 0) {
      console.log('  profiles table exists with', profilesData.length, 'rows (sample)');
      console.log('  Columns:', Object.keys(profilesData[0]));
    } else {
      console.log('  profiles table exists but is empty or no rows returned');
    }
    
    // Try to get table list from information_schema (might not work due to RLS)
    console.log('\n8. Trying to get list of all tables from information_schema:');
    try {
      const { data: tablesData, error: tablesError } = await supabase
        .from('information_schema.tables')
        .select('table_name')
        .eq('table_schema', 'public')
        .limit(10);
      
      if (tablesError) {
        console.log('  Cannot access information_schema due to RLS:', tablesError.message);
      } else if (tablesData) {
        console.log('  Tables in public schema:');
        tablesData.forEach(table => {
          console.log('    -', table.table_name);
        });
      }
    } catch (error) {
      console.log('  Error accessing information_schema:', error.message);
    }
    
    // Check if there's any data in member_balances that might contain the Excel data
    console.log('\n9. Checking member_balances for any financial data:');
    const { data: allBalances, error: allBalancesError } = await supabase
      .from('member_balances')
      .select('*')
      .limit(5);
    
    if (allBalancesError) {
      console.log('  Error fetching member_balances:', allBalancesError.message);
    } else if (allBalances && allBalances.length > 0) {
      console.log('  Found', allBalances.length, 'rows in member_balances:');
      allBalances.forEach((balance, index) => {
        console.log(`  Row ${index + 1}:`, balance);
      });
    } else {
      console.log('  No data found in member_balances table');
    }
    
    // Check if financial_info in members table has any non-zero data
    console.log('\n10. Checking members table for any non-zero financial_info:');
    const { data: membersWithFinancialInfo, error: membersFinError } = await supabase
      .from('members')
      .select('member_number, name, financial_info')
      .not('financial_info', 'is', null)
      .limit(5);
    
    if (membersFinError) {
      console.log('  Error fetching members with financial_info:', membersFinError.message);
    } else if (membersWithFinancialInfo && membersWithFinancialInfo.length > 0) {
      console.log('  Found', membersWithFinancialInfo.length, 'members with financial_info:');
      membersWithFinancialInfo.forEach((member, index) => {
        console.log(`  ${member.member_number} - ${member.name}:`, member.financial_info);
      });
    } else {
      console.log('  No members found with non-null financial_info');
    }
    
  } catch (error) {
    console.error('Unexpected error:', error);
  }
}

checkAllTables().then(() => {
  console.log('\n\nTable check completed.');
  process.exit(0);
}).catch(error => {
  console.error('Script failed:', error);
  process.exit(1);
});